Debugging ts in plugins

kimjoar opened the pull request:
cc @azasypkin

- `npm install` in root,
- `npm install` in `packages/@elastic/kbn-types`, then `npm build` in that folder
- `npm install` in `core_plugins/pid`, then run `npm run build` or `./node_modules/.bin/webpack` in that folder and you'll get the error

```
src/PidPlugin.ts(16,38): error TS2345: Argument of type 'Observable<PidConfig | undefined>' is not assignable to parameter of type 'Observable<PidConfig | undefined>'.
  Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.
```

kimjoar committed:
Make core plugins use `kbn-types`

Commit:
Children of buttons won't capture hover events in firefox (#13015)

* Children of buttons won't capture hover events in firefox

Fixes https://github.com/elastic/kibana/issues/12987

* Make full screen div accessible

* Remove nested aria labels

thomasneirynck committed:
Should only fit on shapes that are part of the result (#12881)

simianhacker committed:
Fixes #12602 - Change TSVB Fields API to use fieldCaps API (#12611)

* Fixes #12602 - Change TSVB Fields API to use fieldCaps API

* Removing obsolete test

* updating tests

* removing the extra sortBy

epixa committed:
Remove translations from management section (#13049)

We need to revisit how we embed translations into templates, so until we
do we might as well go back to embedding text directly in the page. The
translation implementation now makes it difficult to navigate throughout
complicated HTML.

It also suffers from an inability to flag on unused translation keys,
which creates a good deal of technical debt. Close to half of all
translation keys in the translation file in this initial pilot were
already unused since it was introduced a couple of months ago.

jpfxgood committed:
Branching for 6.0: bump version to 7.0.0-alpha1

chrisronline committed:
[Management] Provide a way to fetch index pattern titles (#13030)

* Refactor how `get_ids` works to enable fetching other properties of index patterns, specifically the `title` or name

* Remove unnecessary lines

* Only cache for `id` calls

kimjoar committed:
Move eslint-config-kibana into core (#12725)

* Initial commit

* added actual config

* version 0.0.1

* version 0.0.2

* [no-const-assign] Disallow assignment to const

http://eslint.org/docs/rules/no-const-assign

* [no-redeclare] Disallow redeclaring variables

http://eslint.org/docs/rules/no-redeclare

* version 0.0.3

* [no-unused-vars]: Disallow declaration of variables that are not used in the code.

* Bump to 0.1.0.

* upgrade deps in preperation for babel6 transition

* 0.2.0-alpha1

* use yaml for readability

* 0.2.0

* update/pin peed dependency versions

* 0.2.1

* [quotes] allow template literals

This allows eslint to validate this rule from the styleguide: https://github.com/elastic/kibana/blob/master/style_guides/js_style_guide.md#use-template-strings-to-avoid-escaping-single-quotes

* 0.2.2

* add object-curly-spacing and no-global-assign rules

* sort .eslintrc.yaml rules

* 0.3.0

* add basic react support

* 0.4.0

* Disallow using 'context' in tests

* 0.5.0

* move from .eslintrc.yaml to .eslintrc.js without .json generation (#6)

* Implement import plugin (#7)

* update deps

* include eslint-plugin-import

* Dereference import config (#8)

* reorganize existing rules into groups

* defreference eslint-plugin-import "recommended" config

Based on https://github.com/benmosher/eslint-plugin-import/blob/ea9c92c7324473ef303ac76b127e17af2becd2ee/config/recommended.js

* 0.6.0

* set environment info for import rule

* 0.6.1

* update peerDependencies

* 0.7.0

* Move eslint-config-kibana into packages directory

spalger committed:
override version until elastic/elasticsearch#25876 is merged

chrisronline committed:
Remove premature abstraction (#13042)

Commit:
remove inline awaits (#13043)

* remove inline awaits

* testSubjects.find now always returns a promise, so we need to make sure there is no more inlining

* more find conversions

* need to pass property name

* Make sure async functions are awaited on.

tylersmalley committed:
Updates esArchiver mappings to use single type (#13107)

Signed-off-by: Tyler Smalley <tyler.smalley@elastic.co>

timroes committed:
Plugin installer proxy support (#12753)

Proxy support for plugin installer

Commit:
skip flaky discover test (#13114)

Caused last 4 builds to fail on master.

Will attempt to bring back in
https://github.com/elastic/kibana/pull/13034

cjcenizal committed:
Add linting rules for formatting JSX. (#12810)

* Add linting rules for formatting JSX.
* Bump eslint config version to 0.8.0.

Commit:
ensure timezone attribute in tests is adhered to (#13121)

* ensure timezone attribute is adhered to

Two issues - one the archiver was loading the visualize mapping after
the config setting so it was bring overwritten, and two,
uiSettings.replace wasn’t working correctly to begin with.

* extra line to ensure server is stabilized to prevent transitory failures

tylersmalley committed:
Removed _default_ from esArchiver test and updates data.json (#13122)

I untar'd data.json as it makes it harder to grep/update/review.

Signed-off-by: Tyler Smalley <tyler.smalley@elastic.co>

cjcenizal committed:
Make 'time picker' button in Discover no results prompt keyboard and screen-reader accessible. (#13046)

* Make 'time picker' button in Discover no results prompt keyboard and screen-reader accessible.

lukasolson committed:
Update filter editor docs with addition of advanced option (#13084)

nreese committed:
date_histogram: some bars are too big (#13068)

* handle data sets with holes

* add unit tests

cjcenizal committed:
Upgrade eslint config to 0.8.1. (#13128)

* Fix incorrect peerDependency and bump eslint config version to 0.8.1.
* Upgrade eslint config to 0.8.1. Fix JSX to adhere to new formatting rules.

epixa committed:
esvm: use branch from package.json (#13149)

Since versions are synced across the stack, the esvm config can use the
same branch configuration as Kibana itself.

Commit:
Stabilize and bring back flaky tests (#13034)

* Stabalize tests and bring back flaky test

* Try to fix flaky map radius visualize test

* fix flaky tile map zoom test

* Try to fix flaky zoom test by avoiding zooming in and out with spy panel open

* Remove duplicate wrapped retry around find methods

The retry was moved internally, around a check for stale elements as
well, so this should no longer be neccessary.

azasypkin committed:
Add unit tests the verify correct `dataType` in `jQuery.ajax` request when loading Console app state. (#13092)

thomasneirynck committed:
Save layer setting in the region map UI (#12956)

thomasneirynck committed:
Region map should respect saved center and zoom (#12883)

weltenwort committed:
[eslint-config-kibana] Add jest plugin and settings (#13090)

ppisljar committed:
fix metric default color (#12993)

ppisljar committed:
adding docs for goal and gauge (#13140)

* adding docs for goal and gauge

Commit:
Remove try/catch for short url so the appropriate errors will be propagated to the UI (#13004)

* Remove try/catch for short url so the appropriate errors will be propagated to the UI

* Simply ensure the error is a Boom error by wrapping it, but keep the original error details.

* Boom.wrap can't handle non Error instances, as exist in some of the tests.

* Only support Error objects, and check both status and statusCode

* fix test

* Fix test errors for reals this time

* Break out status and statusCode short url error tests

timroes committed:
Fix typos and punctuation in WMS settings (#13146)

chrisronline committed:
Ensure conflicted fields can be searchable and/or aggregatable (#13070)

* Ensure that conflict fields can be searchable and/or aggregatable in the UI

* Use `some` instead of `reduce`

* Revert UI changes

* Attempt to convert multiple ES types to kibana types, and if they all resolve to the same kibana type, there is no conflict

* Add comma back

* Cleaner code

* Add tests

* Update failing test to handle searchable and aggregatable properly

* Add functional test to ensure similar ES types are properly merged

* Update tests

* Revert shard size

weltenwort committed:
Use version 0.9.0 of eslint-config-kibana (#13177)

spalger committed:
Implement esTestCluster test util (#13099)

* [es/tests] remove unused module

* [testUtil/es] add utility for starting es nodes in tests

* [ftr/tests] use esTestCluster util to start es

* [es/tests/routes] use esTestCluster to start own es

* [testUtils/kbnServer] disable uiSettings unless plugins are enabled

* [testUtils/esTestCluster] use standard test es port by default

* [server/http/tests] add esTestCluster to setup

* [test/config] unify es test config into a single module

* [testUtils/esTestCluster] directory is no longer configurable

* [testUtils/esTestCluster] throw when es.start() is called again without es.stop()

* [testUtils/esTestCluster] is* checks should not mutate state

thomasneirynck committed:
Should respect pinned filters (#13019)

azasypkin commented:
@kjbekkelund two questions: 

1. why does kbn-types need `rxjs` dependency at all? Or at least being peer dependency for `kbn-types` should be enough? Also in this branch `kbn-types` includes `rxjs` as both dependency and peer dependencies (looks like a type). 

2. Shouldn't `rxjs` be peer dependency for `pid` plugin as well? Looks like a perfect use case for peer dependency here? 

`pid` plugin seems to compile successfully if I remove all `rxjs` dependencies from `kbn-types` and make it a peer dependency for `pid` plugin. But not sure if it works if we decide to restrict to only `major` or `minor` version of `rxjs` like `5.x.x` or `5.4.x`...

kjbekkelund commented:
Just kill it. Just me throwing stuff around while trying to debug the brokenness

spalger committed:
[esTestConfig] fix esvm directory (#13190)

simianhacker committed:
Add Checks to Dashboard Context (#13182)

* Fixes #13181 - Check values before adding them to filter

* Adding tests

* Adding check to make sure query is never undefined

spalger committed:
[testUtils/esTestCluster] use more standard api style (#13197)

chrisronline committed:
Update saved objects client usage to `attributes` instead of `_source`. Also, fixing an improper "safe" apply usage. (#13223)

nreese committed:
update spy table headers when columns update (#13130)

* update paginated table headers with columns update

* remove unneeded sleep

* update getDataTableHeaders to preffered format

* use data-test-subj attribute for functional tests instead of CSS selectors

spalger committed:
[server/indexPatterns] expose indexPatternService getter/factory (#13012)

kimjoar committed:
Remove `@elastic` folder from `packages/` (#13089)

Bargs committed:
Add some functional tests for query language switching (#13036)

* Add some functional tests for query language switching

jbudz committed:
[docs] Add docker-compose version to example (#13006)

ppisljar committed:
removes :filters from label (#12962)

ppisljar committed:
in percentage mode tooltip should also show percentages (#13217)

thomasneirynck committed:
Exclude stacktrace from error response of Timelion backend (#12973)

Bargs committed:
Send map filters through  so it get picked up by both kuery and lucene (#13251)

tsullivan committed:
[Framework/Accessibility] Add kuiScreenReaderOnly class (#13133)

* [Framework/Accessibility] Add kuiScreenReaderOnly class

* fix typo

* use GuideLink

* add screen_reader react component and tests

* export KuiScreenReaderOnly

* --wip-- [skip ci]

* fix lint rule

* remove obsolete snapshots

cjcenizal committed:
[UI Framework] Spawn compileCss as a child process to prevent a node-sass fatal error from killing the watch process (#13222)

* Spawn compileCss as a child process to prevent a node-sass fatal error from killing the watch process.
* Document tasks.

thomasneirynck committed:
ensure we are working with data-series (#13266)

This fixes a regression introduced by https://github.com/elastic/kibana/pull/13217.

epixa committed:
Revert "override version until elastic/elasticsearch#25876 is merged" (#13196)

This reverts commit ce8333b70f6034735afc08c4c6792bb1bbc22e89.

spalger committed:
[npm] upgrade postcss-loader (#13279)

Needed to get a version that included https://github.com/postcss/postcss-loader/pull/174

ytzlax committed:
Add Kibana API plugin (#13206)

Bargs committed:
Add Kuery docs and break search page into subsections (#13074)

* Add Kuery docs and break search page into subsections

spalger committed:
[ui/routes] clone defaults before apply (#13295)

* [ui/routes] add failing tests

* [ui/routes] clone defaults before applying them

* Mention issue in related tests

azasypkin committed:
Introduce logging `JsonLayout`. (#13180)

azasypkin committed:
Print `error.stack` for pattern layout and `error.message`, `error.stack` and `error.name` for json layout. (#13309)

cjcenizal committed:
Add AirBnB linting rules for React best practices. Bump to v0.10.0. (#13259)

thomasneirynck committed:
Visualization accessibility issues (#13226)

Commit:
Make sure no one can show the chrome if the default setting is hidden. (#13250)

* Make sure no one can show the chrome if the default setting is hidden.

add tests

Fixes https://github.com/elastic/kibana/issues/13040

* Improve variable and function names and fix tests

jbudz committed:
[visualize spy] Pull index title instead of id (#13254)

* [visualize spy] Pull index title instead of id

* [visualize spy] Check for existence of index before pulling params

tsullivan committed:
[Accessibilty] Make kbn_accessible_click part of autoload/all  (#13299)

* [Accessibility] Ensure kbn-accessible-click is part of autoload/all

* remove unecessary kbn_accessible_click imports

* use an index.js file to expose inner directives

thomasneirynck committed:
Improve region map documentation (#13319)

thomasneirynck committed:
fix typo

lukasolson committed:
Use ui-select for agg selector (#13292)

* Use ui-select for agg selector

* Fix functional tests

* Fix functional tests

* Fix test

* Fix visualize page object

* Use sortPrefixFirst in agg selector

* Fix unit tests

* Fix unit test

* Use not(.ng-hide) rather than last-child

ppisljar committed:
resize was not detected correctly (#13239)

Bargs committed:
Remove simple_query_string hack now that multi_match supports * (#13285)

Bargs committed:
Add documentation link to query bar when Kuery is selected (#13300)

* Add doc link to kuery bar

spalger committed:
Add some testSubject helpers (#13305)

* [testSubjects] add getVisibleTextAll() helper

* [testSubjects] add isSelected() and isSelectedAll() helpers

* [testSubjects/setValue] support wrappers around inputs

* [testSubjects] add isEnabled() helper

* fix typo

Commit:
Fix full screen toggle button when panel is expanded (#13320)

* Add failing tests

* Fix style issues with full screen toggle

sorenlouv committed:
Mock out static files when running in Jest (#13315)

weltenwort committed:
[docs] Add css style guide section on open/closed principle (#12276)

Commit:
Fix accessibility issues with saved object finder (#13152)

* Fix accessibility issues with saved object finder

and use new kui styles

* Dark theme-icy kuiTabs

* Refer to existing dark theme color variables.  Use dark theme hover link color.

* use button instead of div element so no need for kbn-accessible-click

* Add dark theme tab variety to ui framework site, lighten color of background tabs

lukasolson committed:
Remove highlight query (#13231)

* Change use of all_fields in highlight_query to default_field

* Remove highlight query and option

* Fix tests

* Remove unused setting

* Remove lingering references to all_fields

timroes committed:
Move build-info to status page (#13341)

* Move build-info to status page
* Remove info property from management section

azasypkin committed:
Update new-platform dependencies. (#13338)

Commit:
Show full screen option even when write controls are hidden (#13247)

Commit:
Create the concept of embeddableHandlers (#12146)

* Move dashboard panel rendering logic to each registered type.

* Remove dashboard knowledge of click and brush handlers for visualizations

Move it to the VisualizeEmbeddableHandler.

* merge master with manual changes

* No need to use lodash

* Add EmbeddableHandler base class

* Use correct path to embeddable_handlers_registry

* clean up

* Set visualize scope uiState that is of type PersistedState, otherwise it won't actually be set.

* add retry to loading saved search data

* Fix handleError param and remove unnecessary private param

* Rename savePanelState updatePanel and return the new object rather than mutating the original

* Make ContainerAPI a base class and move the dashboard specific functionality into a new class

* Make api's async and clean up documentation

* Fix panel tests

* Fix bug which broke tests - need to pass container-api to dashboard-panel

* Address code comments

- Rename onFilter to addFilter
- Use angular promises instead of async/await
- fix jsdoc
- rename createChildUiState to getInitialState

* changed the wrong variable name

* no need for async or Promise.reject on interface

* add tests that will fail due to spy pane issue in this PR

* Fix logic with spy pane toggle

There is still a bit of a bug here as mentioned in
https://github.com/elastic/kibana/issues/13340 but it will be fixed
separately as it’s also an issue in master

* Fix failing test

Commit:
Use find service instead of remote  (#13296)

* Use find service instead of remote which doesn't have built in safety checks for stale elements

* use data-test-subj instead of id

* add missing function and awaits

* Did not realize WebElements could find descendants

Added some extra functionality for this in find and testSubjects
service, and using in the one place it’s required.

azasypkin committed:
Add support for `Router.post` method and `body` validation. (#13389)

Bargs committed:
Make filter pills keyboard accessible (#13331)

Fixes #12639

It's now possible to tab to each filter's actions and interact with them via the keyboard.

In order to get the actions to show/hide on both mouse hover and action focus I had to create a new filter-pill component that could manage a bit of state to track whether the user was interacting with a given pill or not.

Commit:
Fix dark theme issues with spy toggle and spy pane (#13345)

* Fix dark theme issues with spy toggle and spy pane

* Port dark theme portion of styling kuiIcon into uiFramework library

spalger committed:
[es/healthcheck] ensure that healthcheck stops when server is stopped (#13201)

spalger committed:
[fix/UiSettings] ignore certain errors (#13079)

* [SavedObjectClient] emit detectable errors

* [uiSettingsService] consume new SavedObjectsClient errors

* [SavedObjectsClient] expose errorTypeHelpers as such

* [elasticsearch/tests] recreate error for each test

* [http] wait for elasticsearch plugin to be ready

* [shortUrl/tests] ensure that create request responds with 200

* [shortUrl] use errorTypeHelpers to filter errors

* [uiSettings/savedObjectsClientStub] stub errorTypeHelpers

* [SavedObjectsClient/errors] expose error module so tests can make errors

* [shortUrl/tests] use actual SavedObjectsClient errors

* [uiSettings/savedObjectsClientStub] use actual errors lib

* [SavedObjectsClient] use decorate instead of "wrap"

* [server/routes/uiSettings] refactor routes to forward Boom errors from uiSettings

* [uiSettings] colocate routes and service

* [testUtils/esTestCluster] use more standard api style

* [testUtils/es] add createCallCluster util

* [testUtils/esTestCluster] add getters for client/callCluster

* [es/healthcheck] ensure that healtcheck stops when server is stopped

* [uiSettings/routes] add param/payload validation

* [uiSettings/routes] add tests that verify error behaviors

Commit:
Intentionally show spy pane in full screen mode, but not embed mode (#13395)

Commit:
port k7 popover component over (#13322)

* port k7 popover component over

* Fix line height

* Clean up css

Bargs committed:
Make 'all filters' actions keyboard accessible (#13391)

* Make 'all filters' actions keyboard accessible
* Add extra aria props to actions toggle

lukasolson committed:
Truncate long field names in filter editor (#13379)

* Truncate long field names in filter editor

* Update filter editor styles

Commit:
Be case insensitive when sorting on the visualize and dashboard landing pages (#13397)

spalger committed:
[server/stats] add server.getKibanaStats() for programatic access (#13411)

timroes committed:
Create keyboard mode for ui-ace editor (#13339)

* Add kbn-ui-ace-keyboard-mode directive

* Implemented PR feedback

* Fix broken tests

spalger committed:
[npm] move personal forks into elastic org (#13410)

azasypkin committed:
Use `bodyParser.json()` and `bodyParser.urlencoded() instead of deprecated and vulnerable `bodyParser()`. (#13446)

thomasneirynck committed:
Visualize Accessibility Issues (#13428)

Bargs committed:
Remove esAdmin mention from docs since it no longer exists (#13449)

The angular service was removed in #13000

chrisronline committed:
[Management] Improve accessibility within management (#13364)

* Use header/h tag for tab header, #12861

* Add tabindexes for tables, #12862

* Add for/id pairings, #12863

* Search role, #12868

* Use better descriptions for aria labels, #12865

* Add aria label, #12870

* Button and a tags are already in tab order, removing unnecessary change

* The input field cannot contain the role="search"

* The tabindex needs to be on the element which the click handler

* Remove tabindex and remove the class applying the hover styles - honestly, we can probably make due without any class (using default h2 styles)

* Revert "Remove tabindex and remove the class applying the hover styles - honestly, we can probably make due without any class (using default h2 styles)"

This reverts commit c54ea227889898baef85ee71b3350bbf69ef5d7c.

* Remove the extra code my cat added

* Addressing hidden inputs from #12863

* Remove unnecessary header tag

* Add clarifying comment

* Prevent multiple ids

simianhacker committed:
Fix a bug with the interval label for TSVB (#13440)

* Fix a bug with the interval label for TSVB

* Ensure reduce returns something

lcawl committed:
[DOCS] Update links to Kibana security (#13456)

simianhacker committed:
Fixes #12685 - Improve error handling, add error for missing aggregations key. (#12688)

thomasneirynck committed:
ensure editor mode propagates correctly (#13355)

simianhacker committed:
Moving timezone functions from Timelion to Vis (#13457)

* Moving timezone functions from Timeline to global

* Fixing a missing require

simianhacker committed:
Fixes #13246 - Add checks for empty annotations (#13422)

* Fixes #13246 - Add checkes for empty annotations

* Refactoring get_annotations to use async/await

epixa committed:
remove @bigfunger/jsondiffpatch (#13442)

This dependency is no longer used.

cjcenizal committed:
Fix KuiPopover examples. (#13461)

Commit:
Avoid a wrapped retry and use exists instead of find, which will take up all the time allotted for the retry if it fails once (#13467)

cjcenizal committed:
[UI Framework] Reorganize UI Framework and add Yeoman generator (#13172)

* Reorganize documentation styles so they all live in doc_site/components directory. (#12809)
  - Remove global styles, e.g. body and html element selectors.
* Create global_styles dir with sub-directories. (#12833)
* Add SCSS style guide. (#12850)
* Refactor UI Framework directory structure to house everything in a src directory. (#12880)
  - Add components/index.js and services/index.js files to continue to export JS modules from the root.
* Add KUI Yeoman generator.
* Support creation of components.
* Add documentation generator for main page, demo, and sandbox.
  - Add additional documentation snippets to KUI generator. (#13076)
  - Fix incorrect use of double quotes in KUI generator snippet. (#13086)
  - Remove infrequently used imports from the KUI generator test template. (#13110)
* Mock assets files for Jest. (#13060)
* Fix broken coverage report paths in Jest config. (#13082)
* Update eslint config to lint the new UI Framework directory structure. (#13102)
* Fix positioning of doc site pagination buttons. (#13203)
* Support hasReact prop for sandboxes. (#13270)
* Remove deprecated used of component mixin from KUI generator's SCSS template. (#13377)
* Fix rebasing errors.
  - Add dashboard back to Jest config.
  - Add missing form and tool_bar variables.
* Rename tasks to createComponent and documentComponent.
* Reference correct src paths in README.
* Add children and className to templates' propTypes.
* Add default folder name for page demo.
* Add suffix to sandbox routes.
* Specify testPathIgnorePatterns more clearly.
* Rename component.test.js to test.js so that Jenkins won't try to run it.
* Update npm scripts to depend on local yo dependency, not global.
* Add ui_framework/src to copy task.
* Simplify npm scripts and remove requirement for installing Yeoman from README.
* Add services to moduleNameMapper in jest config.
* Clean up Button and Gallery examples.

cjcenizal committed:
[UI Framework] Add functionality for hiding and showing the chrome when viewing Sandboxes (#13475)

* Update UI Framework doc site with functionality for hiding and showing the chrome when viewing Sandboxes. (#13437)
* Fix formatting in generator-kui templates. (#13443)

cjcenizal committed:
[eslint-config-kibana] Upgrade eslint-config to 0.10.0. (#13323)

* Upgrade eslint-config to 0.10.0.
* Fix linting violations with popover and typography stuff in UI Framework.

lcawl committed:
[DOCS] Use shared versions file for doc builds (#13277)

* [DOCS] Use shared versions file for doc builds

* [DOCS] Address feedback in index-shared1.asciidoc

epixa committed:
github: PR template checklist in a comment

The checklist of items in the PR description should be in a comment so
you don't have to manually remove them on every pull request.

timroes committed:
Initial version of accessibility guide (#13390)

* Initial version of a11y guide

* Move a11y content from HTML guide to a11y guide

* Add PR feedback

* Add PRs feedback

* Add more roles

* Refactor labeling elements section

* Rename "labeling elements" -> "naming elements"

* Use span instead of i for icon examples

* Correct typo

simianhacker committed:
Fixes #12113 - Add timezone to date histogram aggregations for TSVB (#13378)

* Fixes #12113 - Add timezone to date histogram aggregations

* Fixing tests

* Fixing tests

* Moving to universal timezone function

tylersmalley committed:
Prevents canceling index pattern modal from throwing error (#13488)

* Prevents canceling modal from throwing error

Fixes #12818

Signed-off-by: Tyler Smalley <tyler.smalley@elastic.co>

* Prevents refresh and removes returned promise

Signed-off-by: Tyler Smalley <tyler.smalley@elastic.co>

timroes committed:
Improve accessibility of the discover app (#13498)

* Add sectioning to discover app, fix #12633

* Set unique page title for discover app, fix #12641

* Remove sort button if column isn't sortable

Until now the button was just empty, but could still be focused via
keyboard. This is bad accessibility, so the sort button is now removed
completelyw hen the column isn't sortable.

varunsharma27 committed:
#10569 Fixes the position of spy tab buttons at the top of the widget (#13366)

* #10569 Fixes the position of spy tab buttons at the top of the widget

* #10569 Fixes the position of spy tab buttons at the top of the widget.

* Fixes CSS for Spypanel Selector Menu.

lukasolson committed:
Handle booleans in filter editor (#13406)

* Handle booleans in filter editor

* Use select instead of checkbox

BigFunger committed:
Adds getAriaName function and applies it to advanced settings (#13448)

tylersmalley committed:
Prevents default index pattern name when ID is provided (#13353)

Signed-off-by: Tyler Smalley <tyler.smalley@elastic.co>

spalger committed:
remove spammy console.log()

Bargs committed:
Fix fixed-scroll on Discover page (#13516)

Some CSS changes were made in #10592
which caused the fixed-scroll directive to stop working in Discover.
Specifically the overflow rule which was added forced the width and
scrollWidth of the doc-container to be the same, so the fixed-scroll
directive would never add the scroll bar to the page. This PR simply
reverts the changes made in the above linked PR. I retested the original
issues in the latest version of Firefox and was unable to reproduce. It
seems that the original issues may have been fixed in Firefox itself.

Fixes #12416

spalger committed:
[ui/queryBar/tests] attach $elem to DOM so "click" triggers "submit" (#13544)

timroes committed:
Improve Dev Tools accessibility (#13496)

* Make console actions accessible

Use a button isntead of a link will solve all accessibility issues, and
we just need to style the button the same way we style the links.

* Create kbnUiAceKeyboardModeService

This will be needed for situations in which the initialization need to
be started manually, and a simple directive won't be enough.

* Make console input box accessible

* Make editor_actions buttons instead of links

timroes committed:
Improve accessibility of visualizations (#13507)

* Fix markdown vis settings labels

* Fix contrast ratio of markdown blockquotes, #12491

* Fix fontsize of blockquotes in markdown, #12904

* Add vis type description for screen reader, fix #12887

The way tooltips are currently implemented we unfortunately cannot use
the tooltip for the aria-describedby. This is another issue to solve.

* Make main nav a11y when collapsed

* Camel case the vistype id used in HTML

simianhacker committed:
Duration Formatting for TSVB (#13337)

* Adding duration formatter

* Adding decimal places

* Fixing bug with decimals

* Relaxing the duration format test

jbudz committed:
[console] Replace text/plain fallback with application/json (#12294)

* [console] Always set content-type for requests to es

* [console] always set content type to application/json if there is a body

* remove extra space

Bargs committed:
Improves dark theme support in the filter editor (#13522)

Fixes #12955

* Added dark theme support to the ui framework modal component
* Also fixed z-index issue where dashboard panels would appear above the filter editor modal when the user moved or resized the panel (see gif below)

chrisronline committed:
Add aria describes for warning label, #12864 (#13576)

chrisronline committed:
Fix breaking scope changes in the field format editor samples (#13549)

Bargs committed:
Emphasize important difference between Kuery and Lucene query syntax (#13592)

Fixes #13563

Brings additional attention to the fact that Kuery uses implicit ANDs whereas Lucene query syntax uses implicit ORs, which could confuse users if they're not aware of the difference.

simianhacker committed:
Fixes #12866 - Use devicePixelRatio to scale up and down canvas values in TSVB (#13565)

cjcenizal committed:
[UI Framework] [K7] Improve generator snippets (#13598) (#13599)

* Remove extraneous path info from SCSS import snippet.
* Add GuideCode to demo page. Put stateless function params on multiple lines.
* Log an error if you try to create a component with a blank name.

cjcenizal committed:
[Accessibility] Make table pagination controls keyboard-accessible.  (#13541)

* Make table pagination controls keyboard-accessible. Update KuiLink so it can be used on a button element.
* Add additional properties to link mixin to make kuiLink look the same for both buttons and anchor tags.
* Fix tests.

spalger committed:
[savedObjectClient] decorate non-es errors too (#13596)

timroes committed:
Improve timelion accessibility (#13531)

* Fix broken refresh shortcut for non OS X

On non OS X systems, the e.metaKey is most likely the windows key.
Hotkeys with the windows key are usually used as global (non application
internal) hotkeys. Also the documentation states, that Ctrl + Enter
should also work. With this commit it doesn't matter whether you press
the Meta or the Ctrl key + Enter.

* Make timelion datepicker more accessible

* Add meaningful labels to each interval shortcut, and
* Add a proper label to the custom input text field

* Make save options keyboard accessible, fix #12246, fix #12487

* Make sort button of saved-object-finder accessible, fix #12486

* Label timelion options correctly, fix #12907

* Make timelion legends keyboard accessible, fix #11844

* Make timelion autocompletion properly accessible

Fixes #11533, fixes #11525, fixes #12908

* Make timelion charts and actions accessible, fix #12909

* Add some more inline comments

* Fix HTML styles according to styleguide

* Replace sr-only by kuiScreenReaderOnly

* Make timelion autocompletion work with ARIA 1.1

thomasneirynck committed:
fix available metric aggs (#13518)

Bargs committed:
Fix discover sidebar Visualize button for geo_point fields (#13539)

Fixes #12227

It seems the order of aggs is important now. I don't know the underlying cause here, I just saved a good and bad version of tilemap visualizations and compared the saved objects to figure out what changed.

cjcenizal committed:
[UI Framework] Add KuiOutsideClickDetector (#13521)

* Add KuiOutsideClickDetector.
* Convert KuiColorPicker and KuiPopover to use KuiOutsideClickDetector.

cjcenizal committed:
Fix bug with Dashboard breadcrumb alignment. (#13580)

thomasneirynck committed:
remove geo_bounds agg from visualizations (#13536)

* remove geo_bounds agg from visualizations

* remove from horizontal bar chart

rashmivkulkarni committed:
added a helper method-  selectIndexPattern method (#13568)

* added selectIndexPattern method
* removed the additional sleep

spalger committed:
[chrome/nav/lastUrl] do not track redirect routes (#13432)

* [chrome/nav/lastUrl] do not track redirect routes

* [chrome/subUrl] always track subUrl when no router

kimjoar committed:
Upgrade Node + TypeScript (#13609)

* Upgrade to Node 8.4.0

* Upgrade TypeScript

timroes committed:
Show a warning for non accessible charts for screen readers (#13573)

* Show warning for non accessible visualizations, fix 13572

* Mark markdown vis as accessible

Bargs committed:
Disable creation of "exists" filters for scripted fields (#13538)

* Disable exists filter button for scripted fields in doc viewer

* Disabled exists filter button for scripted fields in discover sidebar

* Fix reference to non-existent scope variable

tsullivan committed:
[UI Bundler] Add Aliases for Moment so plugins can use the same instance (#13628)

nreese committed:
Replace "marked" with "markdown-it" (#13623)

* add markdown functional test

* update markdown vis to use markdown-it

* migrate markdown angular filter to markdown-it

* place other uses of marked and remove dependency

* update breaking changes documenation and set linkify to true

nreese committed:
Upgrade kibana to leaflet 1.x (#12367)

* upgrade leaflet modules to 1.x latest

* fix css for leaflet controls

* update draw options so it is easier to see polygon when drawing

* fix chrome touch issues with closing polygon in leaflet draw

* use canvas renderer

* use leaflet-responsive-popup to avoid tooltip cutoff

* remove radius configuration from leaflet heatmap

* make blur a factor of radius. Set maxZoom to map zoom since new values are calculated per precision

* use _.get to avoid error, cannot read property of undefined

* add cluster size slider

* experiments with image differences

* onload not onLoad

* use canvas dimensions

* compare map canvas to stored PNG of map canvas

* remove pixelmatch from project dependecies

* fix broken test - rounding error

* add expected image tests for geohash layers shaded circles and geohash grids

* bump z-index of vis-spy

* update functional test expected data

* update to leaflet 1.2.0

* revert to leaflet 1.0.3 and update expected data set for map functional tests

* test geohash_layer heatmap in unit test

* update region_map functional test since it can not longer pluck map vectors from DOM

* update documentation

izeye committed:
Use "curl -O" for Darwin (#13602)

thomasneirynck committed:
Add interval doc (#13629)

ppisljar committed:
don't hide metric if show labels is false (#13637)

epixa committed:
Shortcut to build asciidocs (#13164)

* Shortcut to build asciidocs

The command to build the Kibana docs using the elastic/docs repo can be
easy to forget, so this script is an easy to way to do it right every
time.

* Configurable docs repo location

For folks that clone the elastic docs repo to a different location than
as a sibling of kibana.

timroes committed:
Remove custom keyboard mode in datepicker (#13644)

tylersmalley committed:
Removes webpack-directory-name-as-main dependency (#13584)

Signed-off-by: Tyler Smalley <tyler.smalley@elastic.co>

kimjoar committed:
Add package lock files

kimjoar committed:
Extract core plugins

kimjoar committed:
Merge branch 'master' into new-platform

kimjoar committed:
Move `getPluginDir` into `Env`

kimjoar committed:
Fix dates in tests (#13744)

kimjoar committed:
Review fixes

kimjoar committed:
Ensure 'target' is removed before re-building plugin

kimjoar committed:
Update package.lock

kimjoar committed:
Run prettier

kimjoar committed:
Fix typo

kimjoar committed:
Remove unnecessary mention of tsconfig.json

kimjoar committed:
Update platform readme
